mac/Keyman4MacIM/Keyman4MacIM/KeySender.m

```diff
@@ -62,20 +56,54 @@ - (void)sendKeyDown:(NSUInteger)keyCode forSourceEvent:(NSEvent *)event includeK
   }
 }
 
--(void) dealloc {
-  CFRelease(self.eventSource);
+- (void)sendBackspaceforEventSource:(CGEventSourceRef)eventSource {
+  [self.appDelegate logDebugMessage:@"KeySender sendBackspaceforSourceEvent"];
+
+  [self postKeyboardEventWithSource:eventSource code:kVK_Delete postCallback:^(CGEventRef eventToPost) {
+      CGEventPost(kCGHIDEventTap, eventToPost);
+  }];
 }
 
-- (void)sendBackspaceforSourceEvent:(NSEvent *)event {
-  [self.appDelegate logDebugMessage:@"KeySender sendBackspaceforSourceEvent"];
-  [self sendKeyDown:kVK_Delete forSourceEvent:event includeKeyUp:YES];
+- (void)postKeyboardEventWithSource: (CGEventSourceRef)source code:(CGKeyCode) virtualKey postCallback:(PostEventCallback)postEvent{
+
+    CGEventRef ev = CGEventCreateKeyboardEvent (source, virtualKey, true); //down
+    if (postEvent)
+        postEvent(ev);
+    CFRelease(ev);
+    ev = CGEventCreateKeyboardEvent (source, virtualKey, false); //up
+    if (postEvent)
+        postEvent(ev);
+    CFRelease(ev);
```

Comment:
This seems like it has a precondition of `if(!postEvent) return;`?
```suggestion
    if(!postEvent) {
      // TODO: add logging? how could this happen?
      return;
    }
    CGEventRef ev = CGEventCreateKeyboardEvent (source, virtualKey, true); //down
    postEvent(ev);
    CFRelease(ev);
    ev = CGEventCreateKeyboardEvent (source, virtualKey, false); //up
    postEvent(ev);
    CFRelease(ev);
```

Reply:
It should never happen, but I'll change to add logging.